docs/fundamentals/messagingcenter.md

```diff
@@ -1,12 +1,124 @@
+   ```xml
+   <ItemGroup>
+     <PackageReference Include="CommunityToolkit.Mvvm" Version="8.*" />
+   </ItemGroup>
+   ```
+
+2. Define a message type (optionally using `ValueChangedMessage<T>` when sending a single payload value):
+
+   ```csharp
+   using CommunityToolkit.Mvvm.Messaging.Messages;
+
+   public sealed class HiMessage : ValueChangedMessage<string>
+   {
+       public HiMessage(string name) : base(name) { }
+   }
+   ```
+
+3. Send a message from the publisher:
+
+   ```csharp
+   using CommunityToolkit.Mvvm.Messaging;
+
+   // Send without a direct reference to any recipients
+   WeakReferenceMessenger.Default.Send(new HiMessage("John"));
+   ```
+
+4. Receive the message in a recipient:
+
+   - Implement `IRecipient<TMessage>` and register with the messenger, or
+   - Inherit from `ObservableRecipient` and enable reception.
+
+   ```csharp
+   using CommunityToolkit.Mvvm.Messaging;
+   using CommunityToolkit.Mvvm.Messaging.Messages;
+
+   public sealed class GreetingViewModel : IRecipient<HiMessage>
+   {
+       public GreetingViewModel()
+       {
+           // Register this instance to receive HiMessage notifications
+           WeakReferenceMessenger.Default.Register(this);
+       }
+
+       public void Receive(HiMessage message)
+       {
+           var name = message.Value; // "John"
+           // Handle the message (e.g., update state)
+       }
+
+       // Call this when the recipient is disposed/no longer needed
+       public void Unregister()
+       {
+           WeakReferenceMessenger.Default.Unregister<HiMessage>(this);
+       }
+   }
+   ```
+
+### Mapping common MessagingCenter patterns
+
+- Send without payload: `MessagingCenter.Send<TSender>(sender, message)` → define an empty message type (no payload) and `Send(new MyMessage())`.
+- Send with payload: `MessagingCenter.Send<TSender, TArgs>(sender, message, args)` → use `ValueChangedMessage<TArgs>` or a custom message with properties.
+- Subscribe callback: `MessagingCenter.Subscribe<TSender>(subscriber, message, callback)` → implement `IRecipient<TMessage>` and register; the `Receive` method is the callback.
+- Unsubscribe: `MessagingCenter.Unsubscribe` → `WeakReferenceMessenger.Default.Unregister<TMessage>(recipient)`.
+
+### When to use events instead
+
+For simple, local communication (for example, between a control and its parent), standard .NET events are often sufficient. To reduce the risk of memory leaks with long-lived publishers, use <xref:Microsoft.Maui.WeakEventManager>:
+
+```csharp
+public class Counter
+{
+    readonly Microsoft.Maui.WeakEventManager _weakEventManager = new();
+
+    public event EventHandler<int> CountChanged
+    {
+        add => _weakEventManager.AddEventHandler(value);
+        remove => _weakEventManager.RemoveEventHandler(value);
+    }
+
+    int _count;
+    public void Increment()
+    {
+        _count++;
+        _weakEventManager.HandleEvent(this, _count, nameof(CountChanged));
+    }
+}
+```
+
+> [!TIP]
```

Comment:
This is not something official, so not sure we want to put it in the official docs, but I also made this as a drop-in replacement: https://github.com/jfversluis/Plugin.Maui.MessagingCenter